README.md
# https-www.pythonanywhere.com-user-piovittoriovigo-consoles-9980624-


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "nodename: glenn-liveweb1"
    5  mkvirtualenv flask-tutorial --python=python3.6
    6  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    7  git status
    8  pip freeze > requirements.txt
    9  git status
   10  git add .
   11  git commit -m "werkzeug-0.14.1 click--6.7"
   12  pip install Flask-Migrate
   13  export FLASK_APP=flask_app.py
   14  flask db init
   15  git status
   16  git add .
   17  git commit -m "/.cache/pip/wheels/15/35/24"
   18  flask db migrate
   19  flask db stamp head
   20  pip freeze > requirements.txt
   21  git status
   22  git add .
   23  git commit -m "reponse a un halbard"
   24  flask db migrate
   25  flask db upgrade
   26  git status
   27  git add .
   28  git commit -m "une passion pour la peinture acrylique"
   29  flask db migrate
   30  flask db upgrade
   31  flask shell
   32  echo "# https-www.pythonanywhere.com-user-piovittoriovigo-consoles-9980624-" >> README.md
   33  git init
   34  git add README.md
   35  git commit -m "first commit"
   36  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-piovittoriovigo-consoles-9980624-.git
   37  git push -u origin master
   38  history
   39  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="piovittoriovigo",
    password="Exeter22",
    hostname="piovittoriovigo.mysql.pythonanywhere-services.com",
    databasename="piovittoriovigo$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "56456432131398798798"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/c98da1f40510_.py
"""empty message

Revision ID: c98da1f40510
Revises: 
Create Date: 2018-08-27 23:30:16.664272

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'c98da1f40510'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/8efb0840c9f3_.py
"""empty message

Revision ID: 8efb0840c9f3
Revises: c98da1f40510
Create Date: 2018-08-27 23:33:27.257984

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '8efb0840c9f3'
down_revision = 'c98da1f40510'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
